### DISCUSSION TITLE
Adding support for set and frozenset objects

### DISCUSSION THREAD

**Discussion by oleg-oleinik** [pyscripter/python4delphi#437] [Q&A] [answered]
Hello,
Do you plan to connect python objects of type set and frozenset to p4d?
How can a Delphi program now iterate over such objects or check for the presence of specific values?
Thank you.

**Comment by pyscripter** [comment, marked as answer]
You can manipulate them using the sequence api.
For instance using VarPyth:

```pascal
PySet := VarPythonEval('set("abc")');
for var member in VarPyIterate(PySet) do
  WriteLn(member);
if PySet.Contains("a") then
  Writeln('PySet contains "a");
```

**Reply by oleg-oleinik** [reply]
Thanks for the detailed answer.

**Comment by pyscripter** [comment]
I have turned this into an issue #438 so that the python set C-API will be added.
